tests/workflows/pw/test_base.py

```diff
@@ -312,7 +312,13 @@ def test_restart_mode(generate_workchain_pw, generate_calc_job_node, restart_mod
     inputs['pw']['parent_folder'] = node.outputs.remote_folder
     inputs['pw']['parameters'] = Dict({'CONTROL': {'restart_mode': restart_mode}})
 
-    process = generate_workchain_pw(inputs=inputs)
+    if restart_mode == 'restart':
+        process = generate_workchain_pw(inputs=inputs)
+    else:
+        with pytest.warns(UserWarning) as warning:
+            process = generate_workchain_pw(inputs=inputs)
+        assert 'but no input parameters were' in str(warning[0])
```

Comment:
you can use the `match` argument in `pytest.warns` to compare the warning to a regex: https://docs.pytest.org/en/6.2.x/warnings.html#warns
Also the typical approach for `pytest.raises`

Reply:
This is what I get for using the free version of ChatGPT. Will fix 👍